ocaml/stdlib/float_u.mli

```diff
@@ -0,0 +1,344 @@
+# 1 "float_u.mli"
```

Comment:
Modules in the stdlib should normally be exposed through the MODULE ALIASES section of stdlib.ml{,i}. You have not done that here. Is that intentional? I could see arguments either way, but actually I think we should expose the module, given our safety check.

You may wish to double-check the instructions in https://github.com/ocaml-flambda/flambda-backend/blob/main/ocaml/stdlib/HACKING.adoc

Reply:
I think the plan was to do what we did for `IArray` (i.e., explicitly not include an alias in the main `stdlib` module).  I'm not actually sure what the reasons to do that in the case of `IArray` are, and agree there isn't a super strong safety reason, but perhaps there were reasons related to public release libraries?

Reply:
@antalsz Can you remind us what the reasoning was for `IArray`?

Reply:
Hmm...... public release is a good reason. Putting in the `Stdlib__` is enough of a speed bump that people won't do this accidentally, which would be annoying in a public-release library. Let's go with that. Would be good to document all this somewhere.

Reply:
I've added a comment.